axelor-account/src/main/resources/views/InvoicePayment.xml

```diff
@@ -46,9 +46,9 @@
         if="__config__.app.getApp('base').getEnableTradingNamesManagement()"/>
       <field name="invoice" hidden="true"/>
     </panel>
-    <panel name="discountPanel" title="Financial Discount"
+    <panel name="discountPanel" title="Financial Discount" showIf="applyFinancialDiscount"
       if="__config__.app.getApp('account').getManageFinancialDiscount()">
-      <field name="applyFinancialDiscount" readonlyIf="!financialDiscount"
+      <field name="applyFinancialDiscount" readonly="true"
         onChange="action-invoice-payment-method-change-amount,action-attrs-on-change-invoice-payment-financial-discount"/>
```

Comment:
please remove "onChange" attribute and the action attrs since the field is readonly